text/0143-sdk-fail-safe-mode.md

```diff
@@ -0,0 +1,512 @@
+    Given there is no start init marker file for the app version
+    And there is no success init marker file for the app version
+    When the SDK reaches the successful init checkpoint
+    Then the SDK creates a success init marker file for the app version
+
+Scenario: App version inits the first time with failed init
+    Given there is no start init marker file for the app version
+    And there is no success init marker file for the app version
+    When the SDK crashes before reaching the successful init checkpoint
+    Then the SDK can't create a success init marker file for the app version
+    And the SDK can't determine a failed init
+
+Scenario: SDK version inits successfully second time with previous successful init
+    Given there is a start init marker file for the app version
+    And there is a success init marker file for the app version
+    When the SDK inits
+    Then it deletes the success init marker file for the app version
+    When the SDK reaches the successful init checkpoint
+    Then the SDK creates a success init marker file for the app version
+
+Scenario: App version inits with failure second time with previous successful init
+    Given there is a start init marker file for the app version
+    And there is a success init marker file for the app version
+    When the SDK inits
+    Then it deletes success init marker file for the app version
+    When the SDK crashes before reaching the successful init checkpoint
+    Then the SDK can't create a success init marker file for the app version
+
+Scenario: App version inits with previous failed init
+    Given there is a start init marker file for the app version
+    And there is no success init marker file for the app version
+    When the SDK inits
+    Then it determines a failed init
+
+Scenario: New App version inits with previous failed init
+    Given there is a start init marker file for a previous app version
+    And there is no success init marker file for the previous app version
+    And there is no success init marker file for the current app version
+    When the new app version inits
+    Then the SDK deletes marker files from the previous app version
+    And the SDK creates a start init marker file for the current app version
+```
+
+As we must access checkpoint information during the application launch, we must choose an efficient way to read and write this information to not slow down our users' apps. Depending on the platform, we can use marker files or key-value stores such as [UserDefaults](https://developer.apple.com/documentation/foundation/userdefaults) on Apple or [SharedPreferences](https://developer.android.com/reference/android/content/SharedPreferences) on Android. Marker files are more efficient than reading file contents because, for these, the OS only needs to check the file's existence, which is usually a system metadata look-up. We still need to determine which approach is the most efficient.
```

Comment:
```suggestion
As we must access checkpoint information during the application launch, we must choose an efficient way to read and write this information to not slow down our users' apps. Depending on the platform, we can use marker files, key-value stores such as [UserDefaults](https://developer.apple.com/documentation/foundation/userdefaults) on Apple or [SharedPreferences](https://developer.android.com/reference/android/content/SharedPreferences) on Android or System APIs like [ApplicationExitInfo on Android](https://developer.android.com/reference/android/app/ApplicationExitInfo). Marker files are more efficient than reading file contents because, for these, the OS only needs to check the file's existence, which is usually a system metadata look-up. We still need to determine which approach is the most efficient.
```